datafusion/src/physical_plan/type_coercion.rs

```diff
@@ -131,7 +131,10 @@ fn get_valid_types(
         Signature::OneOf(types) => {
             let mut r = vec![];
             for s in types {
-                r.extend(get_valid_types(s, current_types)?);
+                if let Ok(valid_types) = get_valid_types(s, current_types) {
```

Comment:
why is this necessary?

Reply:
Maybe I have misunderstood it, but in the original code, if one of the signatures within OneOf fails then the whole get_valid_types returns an error, isn't it ?

Reply:
you can also use [`filter_map`](https://doc.rust-lang.org/rust-by-example/error/iter_result.html#ignore-the-failed-items-with-filter_map) and then take first

Reply:
Thanks ! changed to filter_map.